src/parse.rs

```diff
@@ -2219,34 +2219,35 @@ impl<'a> Parser<'a> {
                                     } else {
                                         ItemBody::Link(link_ix)
                                     };
-                                    let label_node = self.tree[tos.node].next;
-
-                                    // lets do some tree surgery to add the link to the tree
-                                    // 1st: skip the label node and close node
-                                    self.tree[tos.node].next = node_after_link;
-
-                                    // then, if it exists, add the label node as a child to the link node
-                                    if label_node != cur {
-                                        self.tree[tos.node].child = label_node;
-
-                                        // finally: disconnect list of children
-                                        if let TreePointer::Valid(prev_ix) = prev {
-                                            self.tree[prev_ix].next = TreePointer::Nil;
-                                        }
-                                    }
-
-                                    self.tree[tos.node].item.end = end;
-
-                                    // set up cur so next node will be node_after_link
-                                    cur = TreePointer::Valid(tos.node);
-                                    cur_ix = tos.node;
 
                                     if tos.ty == LinkStackTy::Link {
                                         self.link_stack.disable_all_links();
                                     }
                                 } else {
                                     self.tree[cur_ix].item.body = ItemBody::Text;
                                 }
+                                // assume the reference link was meant to be valid
```

Comment:
I don't completely understand what this change does. Is it reordering some operations, and if so, for what reason?

Reply:
The idea, which didn't work, was to treat `[a][b]` as a link even if `b` didn't resolve. Before it would only be treated as a link if the reference was valid. Now it causes the following markdown to be parsed incorrectly:
```
[foo][bar][baz]

[baz]: /url
```
It should be parsed as `[foo]` and then a link to `/url` with text `bar`. Instead it's now parsed as a self-contained reference link with text `bar`. I think this is because I took `[bar]` completely out of the tree so it isn't being considered as a label.